### ISSUE TITLE
Layer flatten does not support masking (Multi-Label Classfication)

### ISSUE THREAD

**Issue by deanhoperobertson** [keras-team/keras#13189]
<em>I am getting this error when trying to apply **mask_zero = True** in the embedding layer and later **Flatten** the dimensions to meet the requirements of output (2D). 

**Sample Code**  
-----------------------------------------------------------------
`input_layer = Input(shape=(MAX_SEQUENCE_LENGTH,),dtype='int32')

embedding_layer = Embedding(input_dim=n_words+1,output_dim=EMBEDDING_DIM,
                            input_length=MAX_SEQUENCE_LENGTH,
                            trainable=True,
                            mask_zero=True)(input_layer)

drop1 = SpatialDropout1D(0.3)(embedding_layer)

lstm_1 = Bidirectional(LSTM(64,recurrent_dropout=0.1,dropout=0.5, 
                            kernel_initializer='glorot_uniform',
                            return_sequences=True),merge_mode='concat')(drop1)

lstm_1 = BatchNormalization()(lstm_1)
att_layer = Flatten()(lstm_1)
drop3 = Dropout(0.5)(att_layer)
predictions = Dense(33, activation='sigmoid')(drop3)
model = Model(inputs=input_layer, outputs=predictions)

-----------------------------------------------------------------
Any idea how to get around this??

**Comment by PhilipMay** [comment]
Can you provide the error message please?

**Comment by deanhoperobertson** [comment]
Error message:

 Layer flatten_2 does not support masking, but was passed an input_mask: Tensor("embedding_2/NotEqual:0", shape=(?, 366), dtype=bool)
